numpy/lib/function_base.py

```diff
@@ -3803,21 +3812,79 @@ def percentile(a, q, axis=None, out=None,
     if not _quantile_is_valid(q):
         raise ValueError("Percentiles must be in the range [0, 100]")
     return _quantile_unchecked(
-        a, q, axis, out, overwrite_input, interpolation, keepdims)
+        a, q, w, axis, out, overwrite_input, interpolation, keepdims)
 
 
-def _quantile_dispatcher(a, q, axis=None, out=None, overwrite_input=None,
-                         interpolation=None, keepdims=None):
-    return (a, q, out)
+def _weighted_ureduce(a, func, w, **kwargs):
+    """
+    Internal Function.
+    Call `func` with `a` and `w` as arguments swapping the axes to use extended
+    axis on functions that don't support it natively.
+    
+    Returns result and a.shape with axis dims set to 1.
 
+    Parameters
+    ----------
+    a : array_like
+        Input array or object that can be converted to an array.
+    func : callable
+        Reduction function capable of receiving a single axis argument.
+        It is called with `a` as first argument followed by `kwargs`.
+    w : array_like
+        It has the sample shape with a.shape.
+    kwargs : keyword arguments
+        additional keyword arguments to pass to `func`.
+
+    Returns
+    -------
+    result : tuple
+        Result of func(a, w, **kwargs) and a.shape and with axis dims set to 1
+        which can be used to reshape the result to the same shape a ufunc with
+        keepdims=True would produce.
+
+    """        
+    axis = kwargs.get('axis', None)
+    if axis is not None:
+        keepdim = list(a.shape)
+        nd = a.ndim
+        axis = _nx.normalize_axis_tuple(axis, nd)
+
+        for ax in axis:
+            keepdim[ax] = 1
 
-@array_function_dispatch(_quantile_dispatcher)
+        if len(axis) == 1:
+            kwargs['axis'] = axis[0]
+        else:
+            keep = set(range(nd)) - set(axis)
+            nkeep = len(keep)
+            # Swap axis that should not be reduced to front
+            for i, s in enumerate(sorted(keep)):
+                a = a.swapaxes(i, s)
+                w = w.swapaxes(i, s)
+            # Merge reduced axis
+            a = a.reshape(a.shape[:nkeep] + (-1,))
+            w = w.reshape(a.shape[:nkeep] + (-1,))
+            kwargs['axis'] = -1
+        keepdim = tuple(keepdim)
+    else:
+        keepdim = (1,) * a.ndim
+
+    r = func(a, w, **kwargs)
+    return r, keepdim
+
+
+def quantile_dispatcher(a, q, axis=None, out=None,
+                        overwrite_input=None, interpolation=None,
+                        keepdims=None, w=None):
+    return (a, q, out, w)
+
+
+@array_function_dispatch(quantile_dispatcher)
 def quantile(a, q, axis=None, out=None,
-             overwrite_input=False, interpolation='linear', keepdims=False):
+             overwrite_input=False, interpolation='linear',
+             keepdims=False, w=None):
     """
-    Compute the q-th quantile of the data along the specified axis.
-
-    .. versionadded:: 1.15.0
```

Comment:
Irrelevant code should not be deleted.

Reply:
I fetched code with older version at first and caused that deletion.
This section is not in the master.

Reply:
@r08922129, it definitely is in master: 

https://github.com/numpy/numpy/blob/a0028bca0117874606bce99261d978df8d3f6610/numpy/lib/function_base.py#L3818-L3820

Reply:
Sorry, I'll check it again